KLR/NRT/leannrt.c

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Govereau, Sean McLaughlin
+*/
+
+// Modified from https://awsdocs-neuron.readthedocs-hosted.com/en/latest/neuron-runtime/nrt-api-guide.html#the-code
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <lean/lean.h>
+#include <nrt/nrt.h>
+#include <nrt/nrt_experimental.h>
+
+#define DEBUG_ENABLED 0
+
+#if DEBUG_ENABLED
+#define DEBUG_PRINTF(...) fprintf(stderr, __VA_ARGS__)
+#else
+#define DEBUG_PRINTF(...) ((void)0)
+#endif
+
+#define P_ERR(...) fprintf(stderr, __VA_ARGS__)
+
+#define FILENAME_MAX_LENGTH 280
+
+lean_object* io_err(const char* msg) {
+  return lean_io_result_mk_error(lean_mk_io_user_error(lean_mk_string(msg)));
+}
+
+// TODO: Add formatting arguments to msg
+#define CHECK_RESULT(result, msg) \
```

Comment:
How can I add printf-style formatting to this? I guess it would need to allocate a new buffer?